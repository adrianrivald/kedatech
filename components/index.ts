export * from "./Layout"
export * from "./Hero/Hero"