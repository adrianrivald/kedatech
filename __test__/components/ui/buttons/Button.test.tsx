import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import renderer from "react-test-renderer";
import { Button } from "../../../../components/Ui";

it("Button snapshot test", () => {
  const tree = renderer
    .create(
      <Button title="Test Button" isPrimary className="w-full lg:w-auto" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Button text inside test", () => {
  render(<Button title="Test Button" isPrimary className="w-full lg:w-auto" />);
  const divElement = screen.getByRole("mainButton");
  expect(divElement).toHaveTextContent("Test Button");
});
