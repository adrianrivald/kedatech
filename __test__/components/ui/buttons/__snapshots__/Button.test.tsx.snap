// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test 1`] = `
<button
  className="rounded-lg px-4 py-2 border-2 min-h-12 bg-primary-500 border-primary-500 text-white w-full lg:w-auto"
  role="mainButton"
>
  Test Button
</button>
`;
